### PR TITLE
added test for long file name

### PR DESCRIPTION
**Pull request by sushmita56**
### Related Issue
#9396 
### Description
The file name with length(>=233 character) is not synced and is blacklisted.

Screenshot from 2022-01-28 15-26-33

### CODE REVIEW

**Comment by ownclouders** [comment]
Results for <strong>GUI-tests</strong> https://drone.owncloud.com/owncloud/client/10800/5/1
:boom: The GUI tests failed.
GUI Logs: (https://cache.owncloud.com/public/owncloud/client/10800/guiReportUpload/index.html)            
Server Logs: ((https://cache.owncloud.com/public/owncloud/client/10800/guiReportUpload/serverlog.log)

File: test/gui/shared/steps/steps.py [hunk 1]
```diff
@@ -327,8 +328,9 @@ def collaboratorShouldBeListed(
 
 @When('the user waits for the files to sync')
 def step(context):
+    snooze(20)
```
Comment (Talank) [inline test/gui/shared/steps/steps.py:331]: I think that static wait without reasonable explanation should be removed. In fact, there should be some change in this step definition. @saw-jan did the change AFAIK.

File: test/gui/shared/steps/steps.py [hunk 2]
```diff
@@ -369,16 +371,14 @@ def createFile(context, filename, username=None):
...
     # https://github.com/owncloud/client/issues/9325
     snooze(5)
-
     f = open(join(syncPath, filename), "w")
     f.write(fileContent)
     f.close()
-
-
+    
+
```
Comment (Talank) [inline test/gui/shared/steps/steps.py:381]: Can you run `black` once again. Try to avoid changes in empty lines.

File: test/gui/shared/steps/steps.py [hunk 3]
```diff
@@ -398,13 +398,47 @@ def createFolder(context, foldername, username=None):
     path = join(syncPath, foldername)
     os.makedirs(path)
 
+ 
+@When('user "|any|" creates "|any|" files each of size "|any|" bytes inside the folder "|any|"')
+def step(context, username, filenumber, filesize, foldername):
+    syncPath = context.userData['currentUserSyncPath']
+    path = join(syncPath, foldername)
+    filesize = builtins.int(filesize)
+    snooze(5)
```
Comment (Talank) [inline test/gui/shared/steps/steps.py:407]: Please include appropriate reason for the static wait.

File: test/gui/shared/steps/steps.py [hunk 4]
```diff
@@ -1333,3 +1380,4 @@ def step(context, username):
         filename = syncPath + row[0]
         f = open(join(syncPath, filename), "w")
         f.close()
+    test.xvp("VP_VFS_enabled")
```
Comment (Talank) [inline test/gui/shared/steps/steps.py:1383]: Why do you need xvp here? FYI: The `xvp` is used to test the visual verification that is expected to fail.

File: test/gui/tst_syncing/test.feature [hunk 5]
```diff
@@ -329,4 +330,97 @@ Feature: Syncing files
...
+        # the length of file name is 224
+        When user "Alice" creates a file "<filename>" with the following content inside the sync folder
+            """
+            test content
+            """
+        And the user waits for file "<filename>" to be synced
+        Then as "Alice" file "<filename>" should exist on the server
+        Examples:
+            | filename                                                                                                                                                                                                                        |
+            | sdkdfjsdfuidjfkdsjfksdjfksdjfksdjfksdjfksdjfkdsjfksdjfkdsjfkdsfjlsdkfjsdkjflksdjfklsdjfksdjfkdsjfkldsjfkldsjfkdsjfksdjfksdjfklsdjfklsdjflksdjflksdjfklsdjfklsdjfksdjfksdjfksdjfksdjfksdfjskdfjksdjfksdjfksdjfksdjfksdwwwwww.txt |
```
Comment (Talank) [inline test/gui/tst_syncing/test.feature:347]: can you use a little bit more readable name?

File: test/gui/tst_syncing/test.feature [hunk 6]
```diff
@@ -329,4 +330,97 @@ Feature: Syncing files
...
+            test contents
+            """
+        When user "Alice" creates a file "file1.odt" with the following content inside the sync folder
+            """
+            test contents
+            """
+        And the user waits for file "file1.txt" to be synced
+        And the user waits for file "file1.odt" to be synced
+        Then as "Alice" file "file1.txt" should exist on the server
+        Then as "Alice" file "file1.odt" should exist on the server
```
Comment (Talank) [inline test/gui/tst_syncing/test.feature:379]:
```suggestion
        And as "Alice" file "file1.odt" should exist on the server
```

File: test/gui/tst_syncing/test.feature (hunk 5 above)
Comment (phil-davis) [reply test/gui/tst_syncing/test.feature:347]: In https://github.com/owncloud/core/blob/master/tests/acceptance/features/apiSharePublicLink1/createPublicLinkShare.feature we have:
`aquickbrownfoxjumpsoveraverylazydogaquickbrownfoxjumpsoveralazydog.txt`

You can write a whole sentence that is readable. You can even explain in the file name `thisIsAVeryLongFileNameToCheckThatItWorks0123456789.txt`

File: test/gui/shared/steps/steps.py (hunk 3 above)
Comment (sushmita56) [reply test/gui/shared/steps/steps.py:409]: Syncing all the created files at once seems to have some problem on syncing, as a result not all created files sync successfully. So once a file is synced the process will have to wait unless the next file is created. Also, it takes at least 5 seconds for a file to begin the sync process. So here we added a snooze of 5 seconds.

Also, the step ```When the user waits for the files to sync``` is not working properly as expected. And probably this PR -> https://github.com/owncloud/client/pull/9442 may solve this issue. So we can’t remove snooze(5) unless the issue is fixed.

**Comment by ownclouders** [comment]
Results for <strong>GUI-tests</strong> https://drone.owncloud.com/owncloud/client/10811/5/1
:boom: The GUI tests failed.
GUI Logs: (https://cache.owncloud.com/public/owncloud/client/10811/guiReportUpload/index.html)            
Server Logs: ((https://cache.owncloud.com/public/owncloud/client/10811/guiReportUpload/serverlog.log)

File: test/gui/shared/steps/steps.py (hunk 1 above)
Comment (saw-jan) [reply test/gui/shared/steps/steps.py:331]: Yeah, I did in this PR https://github.com/owncloud/client/pull/9442

File: test/gui/shared/scripts/bdd_hooks.py [hunk 7]
```diff
@@ -38,7 +38,7 @@ def hook(context):
     DEFAULT_CONFIG = {
         'localBackendUrl': 'https://localhost:9200/',
         'secureLocalBackendUrl': 'https://localhost:9200/',
-        'maxSyncTimeout': 10,
+        'maxSyncTimeout': 60,
```
Comment (Talank) [inline test/gui/shared/scripts/bdd_hooks.py:41]: I understand that we need to increase the timeout. But, must it be 60? cc @saw-jan

**Comment by ownclouders** [comment]
Results for <strong>GUI-tests</strong> https://drone.owncloud.com/owncloud/client/10949/5/1
GUI Logs: (https://cache.owncloud.com/public/owncloud/client/10949/guiReportUpload/index.html)
Server Logs: (https://cache.owncloud.com/public/owncloud/client/10949/guiReportUpload/serverlog.log)

File: test/gui/shared/steps/steps.py [hunk 8]
```diff
@@ -467,8 +468,8 @@ def createFile(context, filename, username=None):
 
     # A file is scheduled to be synced but is marked as ignored for 5 seconds. And if we try to sync it, it will fail. So we need to wait for 5 seconds.
     # https://github.com/owncloud/client/issues/9325
-    snooze(5)
 
+    snooze(5)
```
Comment (saw-jan) [inline test/gui/shared/steps/steps.py:472]: you can undo this change

File: test/gui/shared/steps/steps.py [hunk 9]
```diff
@@ -494,12 +495,48 @@ def createFolder(context, foldername, username=None):
...
+    syncPath = context.userData['currentUserSyncPath']
+    path = join(syncPath, foldername)
+    filesize = builtins.int(filesize)
+    snooze(5)
+    for i in range(0, builtins.int(filenumber)):
+        file_name = f"file{i}.txt"
+        file = open(f'{path}/file{i}', "wb")
+        file.seek(filesize)
+        file.write(b"\0")
+        file.close()
```
Comment (saw-jan) [inline test/gui/shared/steps/steps.py:511]: you can use the `uploadFile` function from below

File: test/gui/tst_syncing/test.feature [hunk 10]
```diff
@@ -327,4 +327,97 @@ Feature: Syncing files
...
+
+    Scenario: Folders with 500 files can sync successfully
+        Given user "Alice" has set up a client with default settings
+        When user "Alice" creates a folder "folder1" inside the sync folder
+        And user "Alice" creates "500" files each of size "1048576" bytes inside the folder "folder1"
+        And the user waits for the files to sync
+        And user "Alice" creates a folder "folder2" inside the sync folder
+        And user "Alice" creates "500" files each of size "1048576" bytes inside the folder "folder2"
+		And the user waits for the files to sync
+        Then as "Alice" folder "folder1" should exist on the server
```
Comment (saw-jan) [inline test/gui/tst_syncing/test.feature:406]:
```suggestion
        And user "Alice" creates "500" files each of size "1048576" bytes inside the folder "folder2"
	And the user waits for the files to sync
        Then as "Alice" folder "folder1" should exist on the server
```

File: test/gui/tst_syncing/test.feature [hunk 11]
```diff
@@ -327,4 +327,97 @@ Feature: Syncing files
...
         And as "Alice" file "test_video.mp4" should exist on the server
-        And as "Alice" file "simple.txt" should exist on the server
\ No newline at end of file
+        And as "Alice" file "simple.txt" should exist on the server
+
+
+    Scenario Outline: File with long name can be synced
+        Given user "Alice" has set up a client with default settings
+        # the length of file name is 228
+        When user "Alice" creates a file "<filename>" with the following content inside the sync folder
```
Comment (saw-jan) [inline test/gui/tst_syncing/test.feature:336]:
```suggestion
    Scenario Outline: File with long name (288 characters) can be synced
        Given user "Alice" has set up a client with default settings
        When user "Alice" creates a file "<filename>" with the following content inside the sync folder
```

File: test/gui/tst_syncing/test.feature [hunk 12]
```diff
@@ -327,4 +327,97 @@ Feature: Syncing files
...
+        And the user waits for file "<filename>" to be synced
+        Then as "Alice" file "<filename>" should exist on the server
+        Examples:
+            | filename                                                                                                                                                                                                                             |
+            | thisIsAVeryLongFileNameToCheckThatItWorks-thisIsAVeryLongFileNameToCheckThatItWorks-thisIsAVeryLongFileNameToCheckThatItWorks-thisIsAVeryLongFileNameToCheckThatItWorks-thisIsAVeryLongFileNameToCheckThatItWorks-thisIsAVeryLon.txt |
+
+
+    Scenario Outline: Filename with long name(>228 character) raise an error
+        Given user "Alice" has set up a client with default settings
+        # the length of file name is 229
```
Comment (saw-jan) [inline test/gui/tst_syncing/test.feature:349]: characters already mentioned in outline description
```suggestion
```

File: test/gui/tst_syncing/test.feature [hunk 13]
```diff
@@ -327,4 +327,97 @@ Feature: Syncing files
...
+            | aQuickBrownFoxJumpsOverAVeryLazyDog-aQuickBrownFoxJumpsOverAVeryLazyDog-aQuickBrownFoxJumpsOverAVeryLazyDog-aQuickBrownFoxJumpsOverAVeryLazyDog-aQuickBrownFoxJumpsOverAVeryLazyDog-aQuickBrownFoxJumpsOverAVeryLazyDog-aQuickBro.txt|
+
+
+    Scenario: Files with same name but different extension sync correctly
+        Given user "Alice" has set up a client with default settings
+        When user "Alice" creates a file "file1.txt" with the following content inside the sync folder
+            """
+            test contents
+            """
+        When user "Alice" creates a file "file1.odt" with the following content inside the sync folder
```
Comment (saw-jan) [inline test/gui/tst_syncing/test.feature:369]:
```suggestion
        And user "Alice" creates a file "file1.odt" with the following content inside the sync folder
```

File: test/gui/tst_syncing/test.feature [hunk 14]
```diff
@@ -327,4 +327,97 @@ Feature: Syncing files
...
+            """
+            test contents
+            """
+        And the user waits for file "file1.txt" to be synced
+        And the user waits for file "file1.odt" to be synced
```
Comment (saw-jan) [inline test/gui/tst_syncing/test.feature:374]: wait for all at once
```suggestion
        And the user waits for the files to sync
```

File: test/gui/tst_syncing/test.feature [hunk 15]
```diff
@@ -327,4 +327,97 @@ Feature: Syncing files
...
+    Scenario: File with spaces in the name can sync
+        Given user "Alice" has set up a client with default settings
+        When user "Alice" creates a file "file with space.txt" with the following content inside the sync folder
+            """
+            test contents
+            """
+        And user "Alice" creates a folder "testfolder" inside the sync folder
+        And user move the file "file with space.txt" to folder "testfolder"
+        And the user waits for folder "testfolder" to be synced
+        Then as "Alice" file "testfolder/file with space.txt" should exist on the server
```
Comment (saw-jan) [inline test/gui/tst_syncing/test.feature:395]: is this scenario to sync or move file with spaces?

File: test/gui/tst_syncing/test.feature [hunk 16]
```diff
@@ -327,4 +327,97 @@ Feature: Syncing files
...
+    Scenario: Folders with 15 subfolders can sync successfully
+        Given user "Alice" has set up a client with default settings
+        When user "Alice" creates a folder "folder1" inside the sync folder
+        And user "Alice" creates a folder "folder2" inside the sync folder
+        And user "Alice" creates "100" subfolders inside the folder "folder1"
+        And user "Alice" creates "100" subfolders inside the folder "folder2"
+        And the user waits for folder "folder1" to be synced
+        Then as "Alice" folder "folder1" should exist on the server
+        And as user "Alice" folder "folder1" should contain "100" items on the server
+        When the user waits for folder "folder2" to be synced
```
Comment (saw-jan) [inline test/gui/tst_syncing/test.feature:421]: I am afraid this will not work as expected.

File: test/gui/tst_syncing/test.feature [hunk 17]
```diff
@@ -327,4 +327,93 @@ Feature: Syncing files
...
+            test content
+            """
+        And the user waits for file "<filename>" to be synced
+        Then as "Alice" file "<filename>" should exist on the server
+        Examples:
+            | filename                                                                                                                                                                                                                             |
+            | thisIsAVeryLongFileNameToCheckThatItWorks-thisIsAVeryLongFileNameToCheckThatItWorks-thisIsAVeryLongFileNameToCheckThatItWorks-thisIsAVeryLongFileNameToCheckThatItWorks-thisIsAVeryLongFileNameToCheckThatItWorks-thisIsAVeryLon.txt |
+
+
+    Scenario Outline: Filename with long name(>228 characters) raise an error
```
Comment (saw-jan) [inline test/gui/tst_syncing/test.feature:346]:
```suggestion
    Scenario Outline: File with long name(>228 characters) is blacklisted
```

File: test/gui/tst_syncing/test.feature [hunk 18]
```diff
@@ -327,4 +327,93 @@ Feature: Syncing files
...
+        And user "Alice" creates a file "file1.odt" with the following content inside the sync folder
+            """
+            test contents
+            """
+        And the user waits for the files to sync
+        Then as "Alice" file "file1.txt" should exist on the server
+        And as "Alice" file "file1.odt" should exist on the server
+
+
+    Scenario: File of 1 GB size sync succefully
```
Comment (saw-jan) [inline test/gui/tst_syncing/test.feature:376]:
```suggestion
    Scenario: File of 1 GB size sync successfully
```

**Comment by sushmita56** [comment]
I will need to analyze the run time of newly added test scenarios but I'm now involved in some high-priority tasks, will be working on this PR later. Or, ask someone to take over.

**Comment by ownclouders** [comment]
Results for <strong>GUI-tests</strong> https://drone.owncloud.com/owncloud/client/11376/6/1
:boom: The GUI tests failed.
GUI Logs: (https://cache.owncloud.com/public/owncloud/client/11376/guiReportUpload/index.html)
Server Logs: (https://cache.owncloud.com/public/owncloud/client/11376/guiReportUpload/serverlog.log)
Stacktrace: (https://cache.owncloud.com/public/owncloud/client/11376/guiReportUpload/stacktrace.log)

**Comment by saw-jan** [comment]
> Results for **GUI-tests** https://drone.owncloud.com/owncloud/client/11376/6/1 boom The GUI tests failed. GUI Logs: (https://cache.owncloud.com/public/owncloud/client/11376/guiReportUpload/index.html) Server Logs: (https://cache.owncloud.com/public/owncloud/client/11376/guiReportUpload/serverlog.log) Stacktrace: (https://cache.owncloud.com/public/owncloud/client/11376/guiReportUpload/stacktrace.log)

AUT crashed. re-started the CI

File: test/gui/tst_syncing/test.feature [hunk 19]
```diff
@@ -327,4 +327,63 @@ Feature: Syncing files
...
+            """
+            test contents
+            """
+        And the user waits for file "file with space.txt" to be synced
+        Then as "Alice" file "file with space.txt" should exist on the server
+
+
+    Scenario: Folders with 500 files can sync successfully
+        Given user "Alice" has set up a client with default settings
+        And the user has created a folder "folder1" with "500" files each of size "1048576" bytes in the file system
```
Comment (saw-jan) [inline test/gui/tst_syncing/test.feature:380]: Comment can be added.
the step creates folder outside of the sync folder.

Other LGTM

File: test/gui/shared/scripts/bdd_hooks.py [hunk 20]
```diff
@@ -34,16 +34,18 @@ def hook(context):
...
         'minSyncTimeout': 5,
         'middlewareUrl': 'http://localhost:3000/',
         'clientConfigFile': '-',
         'clientRootSyncPath': '/tmp/client-bdd/',
+        'tempFolderPath': '/tmp/client-bdd/temp/',
```
[the remark this answers is not in the document]
Comment (individual-it) [reply test/gui/shared/scripts/bdd_hooks.py:48]: https://docs.python.org/3/library/tempfile.html#tempfile.gettempdir

**Review by Talank — COMMENTED** [review]
LGTM

**Review by individual-it — APPROVED** [review]
